### ISSUE TITLE
Ruffle fail to load any SWF in Waterfox Current

### ISSUE THREAD

**Issue by Xial** [ruffle-rs/ruffle#1643]
Hello,

I am unable to load any flash content with Ruffle when using Waterfox Current.
I am using the _Nightly 2020-11-20 preview_ build, and on most sites that I visit that uses Flash, I get the following error stack:

```
# Error Info
Error name: Error
Error message: panicked at 'pattern creation success: JsValue(InvalidStateError: An attempt was made to use an object that is not, or is no longer, usable
699/Xe<@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:23197
q/<@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:19028
__wbg_createPattern_724bbee1dc7fbc06@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/ruffle.js:1:37012
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3395]:0x2ea6d8
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[432]:0x1208ae
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[845]:0x1d262d
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[435]:0x122e59
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[275]:0x35023
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1077]:0x20dce6
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[716]:0x1a83e2
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[326]:0xbbc8f
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[550]:0x164786
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1523]:0x25c8cc
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[2655]:0x2c873a
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[2404]:0x2b7323
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[4078]:0x2f5ba2
E@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:18573
f@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:17376
)', render/canvas/src/lib.rs:1307:30

Error stack:

$@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:20276
__wbg_new_df6e6ab7a65c4c4d@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/ruffle.js:1:31347
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1201]:0x22757d
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1358]:0x2436eb
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3589]:0x2eee47
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[2813]:0x2d15fd
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3884]:0x2f3a65
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3913]:0x2f3f9a
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3789]:0x2f2642
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3914]:0x2f3fd0
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[3320]:0x2e8669
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[432]:0x120f53
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[845]:0x1d262d
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[435]:0x122e59
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[275]:0x35023
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1077]:0x20dce6
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[716]:0x1a83e2
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[326]:0xbbc8f
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[550]:0x164786
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[1523]:0x25c8cc
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[2655]:0x2c873a
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[2404]:0x2b7323
@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/85f5e4a29d83ee43a391.module.wasm:wasm-function[4078]:0x2f5ba2
E@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:18573
f@moz-extension://df551c0c-3ec8-4374-873c-ea099493a45b/dist/core.ruffle.js:1:17376


# Player Info
Allows script access: true
Player type: Object
SWF URL: sunshine/all.swf
Param movie: sunshine/all.swf
Param quality: high
Param menu: false
Param bgcolor: #FFFFFF
Attribute 0: [object Attr]
Attribute 1: [object Attr]
Attribute 2: [object Attr]
Attribute 3: [object Attr]
Attribute 4: [object Attr]
Attribute 5: [object Attr]

# Page Info
Page URL: https://www.ferryhalim.com/orisinal/

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0
OS: Win32

# Ruffle Info
Ruffle version: 0.1.0
Ruffle source: undefined
```

The same build of this add-on works in Microsoft Edge on the same site on the same computer.
![ruffle-has-edges](https://user-images.githubusercontent.com/31123/99883248-006e6880-2bdb-11eb-9be0-513a8170f43d.png)

I am not certain if it is an issue with Waterfox using an older Firefox ESR at its heart, even in its Current branch, or if it is something else.

Thank you.

**Comment by Herschel** [comment]
There are users in our Discord that said Ruffle works in Firefox; anyone familiar with Waterfox have an idea? It is not on our list of supported browsers, and I am unfortunately not familiar with it so I'm not sure what it supports.

**Comment by ROBERT-MCDOWELL** [comment]
waterfox is based on gecko so it should work.

**Comment by Xial** [comment]
I believe I have found the gotcha that applies to this situation.

Waterfox "Current" is based on the Firefox 68 ESR branch. In the past few days, Waterfox Generation 3 was released, but not automatically rolled out to users. It also was not available and visible as an update when you open the About Waterfox screen in said browser, meaning that the update is _invisible_ to users unless they visit the Waterfox website. 

Generation 3 of Waterfox is based on the Firefox 78 ESR branch, where it seems that the Webassembly components needed for Ruffle are now available and active as part of the browser, but weren't in Generation 2 of Waterfox. It's also safe to say that Waterfox Classic users are up a creek, as that is based on yet an older ESR in the 50s range.

So, my solution/workaround for this is _manually install the latest version_ of Waterfox from its website.
